# This is a BitKeeper generated diff -Nru style patch.
#
# ChangeSet
#   2005/01/14 16:07:58-05:00 davej@redhat.com 
#   [CPUFREQ] gx-suspmod: fix gx_suspmod_get
#   
#   Fix to return a proper cpu speed. it is caused by mistaking the meaning
#   of on and off.
#   
#   Signed-Off-by: Hiroshi Miura <miura@da-cha.org>
#   Signed-off-by: Dominik Brodowski <linux@brodo.de>
#   Signed-off-by: Dave Jones <davej@redhat.com>
# 
# arch/i386/kernel/cpu/cpufreq/gx-suspmod.c
#   2005/01/14 16:07:49-05:00 davej@redhat.com +1 -1
#   [CPUFREQ] gx-suspmod: fix gx_suspmod_get
#   
#   Fix to return a proper cpu speed. it is caused by mistaking the meaning
#   of on and off.
#   
#   Signed-Off-by: Hiroshi Miura <miura@da-cha.org>
#   Signed-off-by: Dominik Brodowski <linux@brodo.de>
#   Signed-off-by: Dave Jones <davej@redhat.com>
# 
diff -Nru a/arch/i386/kernel/cpu/cpufreq/gx-suspmod.c b/arch/i386/kernel/cpu/cpufreq/gx-suspmod.c
--- a/arch/i386/kernel/cpu/cpufreq/gx-suspmod.c	2005-01-15 17:17:58 -08:00
+++ b/arch/i386/kernel/cpu/cpufreq/gx-suspmod.c	2005-01-15 17:17:58 -08:00
@@ -209,7 +209,7 @@
 	if ((gx_params->pci_suscfg & SUSMOD) == 0) 
 		return stock_freq;
 
-	return (stock_freq * gx_params->on_duration) 
+	return (stock_freq * gx_params->off_duration) 
 		/ (gx_params->on_duration + gx_params->off_duration);
 }
 
